components/power_agent/power_agent.py

```
@@ -0,0 +1,577 @@
+            handle = pynvml.nvmlDeviceGetHandleByIndex(gpu_idx)
+            default_mw = pynvml.nvmlDeviceGetPowerManagementDefaultLimit(handle)
+            pynvml.nvmlDeviceSetPowerManagementLimit(handle, default_mw)
+            logger.info(
+                "Restored GPU %d to default TGP (%d W)", gpu_idx, default_mw // 1000
+            )
+        except Exception as e:
+            logger.exception("Failed to restore TGP on GPU %d: %s", gpu_idx, e)
+    try:
+        pynvml.nvmlShutdown()
+    except Exception:
+        pass
```

Ruff (0.15.12)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)